corehq/apps/case_importer/tests/test_importer.py

```diff
@@ -436,6 +438,81 @@ def test_columns_and_rows_align(self):
+        dimagi = make_loc('dimagi', 'Dimagi', self.domain, loc_type.code)
+        inc = make_loc('inc', 'Inc', self.domain, loc_type.code, parent=dimagi)
+        dsi = make_loc('dsi', 'DSI', self.domain, loc_type.code, parent=dimagi)
+        dsa = make_loc('dsa', 'DSA', self.domain, loc_type.code, parent=dimagi)
+
+        with restrict_user_to_location(self, dsa):
+            res = self.import_mock_file([
+                ['case_id', 'name',             'owner_id'  ],  # row 1
+                ['',        'Leonard Nimoy',    inc.group_id],  # row 2 should fail
+                ['',        'Kapil Dev',        dsi.group_id],  # row 3 should fail
+                ['',        'Quinton Fortune',  dsa.group_id],  # row 4 should succeed
+            ])
+
+        case_ids = self.accessor.get_case_ids_in_domain()
+        cases = {c.name: c for c in list(self.accessor.get_cases(case_ids))}
+        self.assertEqual(cases['Quinton Fortune'].owner_id, dsa.group_id)
+        self.assertTrue(res['errors'])
+        error_message = exceptions.InvalidLocation.title
+        error_col = None
+        self.assertEqual(res['errors'][error_message ][error_col]['rows'], [2, 3])
+
+    def test_user_can_access_owner(self):
+        loc_type = LocationType.objects.create(domain=self.domain, name='loc_type', shares_cases=True)
+        dimagi = make_loc('dimagi', 'Dimagi', self.domain, loc_type.code)
+        inc = make_loc('inc', 'Inc', self.domain, loc_type.code, parent=dimagi)
+        dsi = make_loc('dsi', 'DSI', self.domain, loc_type.code, parent=dimagi)
+        dsa = make_loc('dsa', 'DSA', self.domain, loc_type.code, parent=dimagi)
+
+        inc_owner = CommCareUser.create(self.domain, 'inc', 'pw', location=inc)
+        dsi_owner = CommCareUser.create(self.domain, 'dsi', 'pw', location=dsi)
+        dsa_owner = CommCareUser.create(self.domain, 'dsa', 'pw', location=dsa)
+
+        with restrict_user_to_location(self, dsa):
+            res = self.import_mock_file([
+                ['case_id', 'name',             'owner_id'   ],  # row 1
+                ['',        'Leonard Nimoy',    inc_owner._id],  # row 2 should fail
+                ['',        'Kapil Dev',        dsi_owner._id],  # row 3 should fail
+                ['',        'Quinton Fortune',  dsa_owner._id],  # row 4 should succeed
+            ])
+
+        case_ids = self.accessor.get_case_ids_in_domain()
+        cases = {c.name: c for c in list(self.accessor.get_cases(case_ids))}
+        self.assertEqual(cases['Quinton Fortune'].owner_id, dsa_owner._id)
+        self.assertTrue(res['errors'])
+        error_message = exceptions.InvalidLocation.title
+        error_col = None
+        self.assertEqual(res['errors'][error_message ][error_col]['rows'], [2, 3])
+
 
 def make_worksheet_wrapper(*rows):
     return WorksheetWrapper(make_worksheet(rows))
+
+
+@contextmanager
+def restrict_user_to_location(test_case, location):
```

Comment:
I like this API!  There's also a `restrict_user_by_location` test util that you could use to create the role

Reply:
`setUp()` and `tearDown()` methods are only a thing because JUnit didn't have Pythonic context managers. :smirk: 

Thanks for the `restrict_user_by_location` suggestion. I didn't know about that ... but it doesn't delete the roles it creates.  I might change it to use `get_or_create_with_permissions()`.

Reply:
The test class where it's used makes and deletes a domain, and I believe the domain delete cleans up roles and most other things associated with the domain.  Might not work here, but figured it's worth checking for the opportunity to deduplicate.